### PR TITLE
Add scupad support

### PR DESCRIPTION
**Pull request by maleehaakbarRS**
1. Update rs_virgo.repl and add support for pad controller.
2. Add board file for I/O connections
3. For now, LEDs and Buttons are connected on pad I/Os but can be changed as per requirement.

### CODE REVIEW

File: src/Infrastructure
Comment (jorge-rs) [inline src/Infrastructure:1]: These must not be included as part of the PR. Those changes belong to the other repository. Please delete it from this review.

File: platforms/cpus/rs_virgo.repl [hunk 1]
```diff
@@ -44,19 +40,72 @@ spi: Memory.MappedMemory @ sysbus 0xA0800000
...
+//main function I/Os are in range 16-31
+    16 ->gpio@0
+    17 ->gpio@1
+    26 ->gpio@2
+    27 ->gpio@3
+    28 ->gpio@4
+    29 ->gpio@5
+    30 ->gpio@6
+    31 ->gpio@7
+//fpga_gpio I/Os are in range 32-47
```
Comment (jorge-rs) [inline platforms/cpus/rs_virgo.repl:94]: Probably you can increase this comment by mentioning that FPGA GPIO mode is currently supported by using an ATCGPIO100.

File: platforms/cpus/rs_virgo.repl [hunk 2]
```diff
@@ -23,11 +23,7 @@ flash: Memory.MappedMemory @ sysbus 0xB0000000
 // =======================
 button1: Miscellaneous.Button @ sysbus
```
Comment (jorge-rs) [inline platforms/cpus/rs_virgo.repl]: Delete this button definition, as all buttons and leds are expected to be in the rs_virgo-eval.repl

File: platforms/cpus/rs_virgo.repl [hunk 3]
```diff
@@ -23,11 +23,7 @@ flash: Memory.MappedMemory @ sysbus 0xB0000000
...
     invert: false
-    -> scu@2
-
-scu: Miscellaneous.RS_SystemControlUnit @ sysbus 0xA0110000
```
Comment (jorge-rs) [inline platforms/cpus/rs_virgo.repl:28]: Why is the scu model no longer part of the .repl file?

**Review by jorge-rs — CHANGES_REQUESTED** [review]
Please check my comments.

File: platforms/cpus/rs_virgo.repl (hunk 2 above)
Comment (maleehaakbarRS) [reply platforms/cpus/rs_virgo.repl]: this button defination is already a part of master_rs for testing scu . So , should I remove it ?

File: src/Infrastructure
Comment (maleehaakbarRS) [reply src/Infrastructure:1]: This repository is submodule that also gets updated. As I also observe similar case in public repository and our previous PRs , so may be there is no need to remove it. You can check the following link for reference.
https://github.com/renode/renode/pull/550

File: platforms/cpus/rs_virgo.repl (hunk 2 above)
Comment (jorge-rs) [reply platforms/cpus/rs_virgo.repl]: Yes, please

File: src/Infrastructure
Comment (jorge-rs) [reply src/Infrastructure:1]: I see the point, bu tin our case we want to handle this separately from now onwards, so no need to update the submodule

**Review by jorge-rs — APPROVED** [review]

**Pull request merged**